lib/Access.php

```diff
@@ -606,25 +606,39 @@ public function dn2ocname($fdn, $ldapDisplayName = null, $isUser = true) {
 		// outside of core user management will still cache the user as non-existing.
 		$originalTTL = $this->connection->ldapCacheTTL;
 		$this->connection->setConfiguration(['ldapCacheTTL' => 0]);
-		// FIXME DI User and Group Managers
-		if (($isUser && !\OC::$server->getUserManager()->userExists($intName))
+		if (($isUser && $this->shouldMapToUsername($intName))
 			|| (!$isUser && !\OC::$server->getGroupManager()->groupExists($intName))) {
```

Comment:
I agree. Doing it for users but not for groups really seems inconsistent. So, why was the logic to prevent collisions added in the first place? The commit that introduced the random number is https://github.com/owncloud/user_ldap/commit/7da7c63898740a0c70bd73a70f4211813f90ead4

Which goes back to core https://github.com/owncloud/core/pull/1604

That PR speaks volumes:
> [I have no idea what you are doing there but it looks good!](https://github.com/owncloud/core/pull/1604#issuecomment-13402989)

> [I don't fully understand it but you know what you are doing. ;-)](https://github.com/owncloud/core/pull/1604#issuecomment-13428675

> [hehe ;) brief explanation: improved way to make ensure unique (internal) names for LDAP users and groups](https://github.com/owncloud/core/pull/1604#issuecomment-13428430)

To shed some light: the code used to take the configured userid attribute eg uid or email, which might result in collisions. So the chosen solution was to invent a new unique username to differentiate the users in ownCloud. I would argue that the uuid or another uniqe attribute must be used. If collisions apply: don't log in the user and fail hard. Which is what this PR does.

Regarding sync: sync should log an error and continue with the next user.

Reply:
This is pure gold.